package-res/resources/web/prompting/components/DojoDateTextBoxComponent.js

```diff
@@ -77,6 +77,22 @@ define(['cdf/components/BaseComponent', "dojo/date/locale", 'dijit/form/DateText
         },
 
         /**
+         * Parses the date using the correct formatter
+         *
+         * @param {String} value The String with the date to parse
+         * @returns {Date} The parsed string as a Date Object, if it is valid
+         * @private
+         */
+        _parseDate: function (value) {
+          if (this.transportFormatter) {
+            value = this.transportFormatter.parse(value);
+          } else if (this._isLegacyDateFormat()) {
+            value = this.localeFormatter.parse(value, {datePattern: this.dateFormat, selector: "date"});
+          }
+          return value;
```

Comment:
Do we need to throw an exception if no parsing is done, aka when there is no transportFormatter and _isLegacyDateFormat is false?